server/src/main/java/org/elasticsearch/index/query/CoordinatorRewriteContext.java

```diff
@@ -58,6 +58,7 @@ boolean hasTimestampData() {
 
     @Nullable
     public MappedFieldType getFieldType(String fieldName) {
+        // TODO use time series id generation?
```

Comment:
It'd be complex, but we already have the time series id generation code on the coordinating node. So - if you are querying all of the dimensions - we could generate the time series id that must match here and then use the routing it makes to limit the nodes we land on.